scripts/o3de/o3de/manifest.py

```diff
@@ -760,11 +837,45 @@ def get_repo_path(repo_uri: str, cache_folder: str or pathlib.Path = None) -> pa
     return cache_file
 
 
+def get_most_compatible_gem(gem_name: str, 
+                            gem_json_data_by_name: dict or None) -> pathlib.Path or None:
+    """
+    Optimized version of get_most_compatible_object() for gems when we have already
+    opened all the gem.json files
+    :param gem_name The gem name with optional version specifier, example: o3de>=1.2.3
+    :param gem_json_data_by_name Gem data from get_gems_json_data_by_name()
+    """
+    gem_name_with_version_specifier = gem_name
+    gem_name, version_specifier = utils.get_object_name_and_optional_version_specifier(gem_name)
+    if not gem_name in gem_json_data_by_name:
+        return None
+
+    matching_paths = deque()
+    most_compatible_version = Version('0.0.0')
+    for gem_json_data in gem_json_data_by_name[gem_name]:
```

Comment:
We should probably use `gem_json_data_by_name.get(gem_name, {})` to be on the safe side